packages/app/src/cli/services/webhook/send-app-uninstalled-webhook.test.ts

```diff
@@ -0,0 +1,90 @@
+import {sendUninstallWebhookToAppServer} from './send-app-uninstalled-webhook.js'
+import {triggerLocalWebhook} from './trigger-local-webhook.js'
+import {afterEach, beforeEach, describe, expect, it, vi} from 'vitest'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {FetchError} from '@shopify/cli-kit/node/http'
+import {Writable} from 'stream'
+
+beforeEach(async () => {
+  vi.mock('@shopify/cli-kit/node/api/partners')
+  vi.mock('./trigger-local-webhook.js')
+  // Stub out sleep so that we don't have to wait for the retries
+  vi.mock('@shopify/cli-kit/node/system')
+})
+
+afterEach(async () => {
+  vi.clearAllMocks()
+})
+
+const apiVersionsResponse = {
+  publicApiVersions: ['2022', '2023', 'unstable'],
+}
+const webhookSampleResponse = {
+  sendSampleWebhook: {
+    samplePayload: '{ "sampleField": "SampleValue" }',
+    headers: '{ "header": "Header Value" }',
+    success: false,
+    userErrors: [
+      {message: 'Error 1', fields: ['field1']},
+      {message: 'Error 2', fields: ['field1']},
+    ],
+  },
+}
+
+describe('sendUninstallWebhookToAppServer', () => {
```

Comment:
It would be nice to check a couple of other things:
- That the API call to get the Sample is using the expected API version (2023)
- That we are writing to stdout the expected messages

Reply:
I'm usually a bit iffy about testing messages because they make the tests a bit brittle, but I think this is unlikely to change so we can do it.

Reply:
To make it more solid, I like to use a regex to check the main part of the message instead of a full comparison. For example, `/retrying/` instead of `App isn't responding yet, retrying in 5 seconds`.